### DISCUSSION TITLE
[Proposal]: Omitted generic parameters when accessing constants defined in a generic type

### DISCUSSION THREAD

**Discussion by Flithor** [dotnet/csharplang#8586] [Language Ideas]
When accessing a constant defined in a generic type, we must definate type parameters, but constants are stateless, and cannot be override or reset.
We should not need to input type parameters when accessing the constant value ​​defined in generic type.
For example:
```csharp
public class MyGenericType<T1, T2, T3>
{
    public const string MY_CONST_STRING = "Constant String Content";
}
```
Now we should write:
```csharp
var getConstString = MyGenericType<Type1,Type2,Type3>.MY_CONST_STRING;
```
I'd like:
```csharp
var getConstString = MyGenericType<,,>.MY_CONST_STRING;
//or
var getConstString = MyGenericType.MY_CONST_STRING;
```

My only concern is whether this breaks any basic c# syntax conventions?

**Comment by HaloFour** [comment]
The existing generic inference proposals wouldn't cover this case as there are nothing to infer.  I, personally, don't see the language wading into generic wildcards for this specific use case.

I'd suggest writing a separate non-generic class for that constant:

```cs
public static class MyGenericType
{
    public const string MY_CONST_STRING = "Constant String Content";
}
```

**Reply by TahirAhmadov** [reply]
If it's desirable for the constant to be in scope in the generic class, then perhaps the non-generic class can be "instance" (not `static`) and be made the base class of the generic class; then both in-scope and public constant access scenarios are convenient.

**Reply by CyrusNajmabadi** [reply]
Or use a `using static`.
